File: src/exam-core/src/main/java/com/wcpdoc/exam/core/entity/Course.java
package com.wcpdoc.exam.core.entity;

import java.util.Date;
import java.util.List;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.wcpdoc.core.mybatis.IntTypeHandler;

import lombok.Data;

/**
 * 课程实体
 * 
 * v1.0 zhanghc 2026-01-12 10:03:53
 */
@Data
@TableName(value = "EXM_COURSE", autoResultMap = true)
public class Course {
	@TableId(type = IdType.AUTO)
	private Integer id;
	private String name;
	private String content;
	@TableField(typeHandler = IntTypeHandler.class)
	private List<Integer> orgIds;
	@TableField(typeHandler = IntTypeHandler.class)
	private List<Integer> userIds;
	private Integer shareAuth;
	private Integer state;
	private Integer createUserId;
	private Integer updateUserId;
	private Date updateTime;
}

File: h5/src/ts/course/course.ts
// 课程接口
export interface Course {
    id: number | null; // ID
    name: string; // 名称
    content: string; // 内容
    orgIds: number[]; // 机构IDS
    userIds: number[]; // 用户IDS
    shareAuth: number | null // 共享权限（0：私有；1：只读；2：读写；）
    state: number | null; // 状态（0：删除；1：发布；2：暂停）
}
